apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/ListBucketsTest.cs

```diff
@@ -75,6 +75,26 @@ public void PartialResponses()
             }
         }
 
+        [Fact]
+        // Fetches soft deleted buckets using the list bucket option with soft delete true.
+        public async Task ListSoftDeletedBuckets()
+        {
+            var actualBuckets = await _fixture.Client.ListBucketsAsync(_fixture.ProjectId, new ListBucketsOptions { SoftDeleted = true }).ToListAsync();
+            
+            foreach (var bucket in actualBuckets)
+            {
+                // Check if list contains only soft deleted buckets created in storage fixture
```

Comment:
This is not what you are checking here. You are only saying to tests assertions on buckets whose names en with soft-delete.

Ain't there a metadata field indicating that the bucket has been softdeleted?

Reply:
SoftDeleted option is there for ListBucketsOptionsto check if bucket has been softdeleted or not.